frontend/src/metabase/containers/CollectionMoveModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from "react";
+
+import { t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import ModalContent from "metabase/components/ModalContent";
+
+import CollectionPicker from "metabase/containers/CollectionPicker";
+import CreateCollectionModal from "metabase/collections/containers/CreateCollectionModal";
+import { Collection, CollectionId } from "metabase-types/api";
+import { ButtonContainer } from "./CollectionMoveModal.styled";
+
+interface CollectionMoveModalProps {
+  title: string;
+  onClose: () => void;
+  onMove: (collection: any) => void;
+  initialCollectionId?: number | string | null;
+}
+
+export const CollectionMoveModal = ({
+  title,
+  onClose,
+  onMove,
+  initialCollectionId,
+}: CollectionMoveModalProps) => {
+  // will eventually be the collection object representing the selected collection
+  // we store the whole object instead of just the ID so that we can use its
+  // name in the action button, and other properties
+  //
+  //  undefined = no selection
+  //  null = root collection
+  //  number = non-root collection id
+  //
+  const [selectedCollectionId, setSelectedCollectionId] =
+    useState(initialCollectionId);
+
+  // whether the move action has started
+  // TODO: use this loading and error state in the UI
+  // const [moving, setMoving] = useState(false);
+  // const [error, setError] = useState(null);
+
+  const [creatingCollection, setCreatingCollection] = useState(false);
+  const [openCollectionId, setOpenCollectionId] =
+    useState<CollectionId>("root");
+
+  if (creatingCollection) {
+    return (
+      <CreateCollectionModal
+        collectionId={openCollectionId}
+        onClose={() => setCreatingCollection(false)}
+        onCreate={(collection: Collection) => {
+          onMove({ id: collection.id });
+        }}
+      />
+    );
+  }
+
+  return (
+    <ModalContent title={title} onClose={onClose}>
+      <CollectionPicker
+        initialOpenCollectionId={openCollectionId}
+        onOpenCollectionChange={setOpenCollectionId}
+        value={selectedCollectionId}
+        onChange={setSelectedCollectionId}
+      />
+      <ButtonContainer>
+        <Button light icon="add" onClick={() => setCreatingCollection(true)}>
+          {t`New collection`}
+        </Button>
+        <Button
+          primary
+          className="ml-auto"
+          disabled={
+            selectedCollectionId === undefined ||
+            selectedCollectionId === initialCollectionId
+          }
+          onClick={() => {
```

Comment:
```suggestion
          onClick={() => onMove({ id: selectedCollectionId })}
```

Following up on the above comment, let's drop the try/catch/finally and commented code.